src/widget/scroll.rs

```diff
@@ -139,6 +147,49 @@ impl<T: Data, W: Widget<T>> Scroll<T, W> {
         self.scroll_offset
     }
 
+    fn calc_scrollbar_screen_edges(&self, viewport: &Rect) -> Rect {
```

Comment:
total nit: you pass `viewport` by reference here, but by value elsewhere. I'd probably just always pass it by value?

Reply:
Ahhh yes, that's a good point